### PR TITLE
LESS mixin fix

### PR DESCRIPTION
**Pull request by HanabishiRecca**
# Description

Fixes Issue: 
https://github.com/beautify-web/js-beautify/issues/1018
https://github.com/beautify-web/js-beautify/issues/1582
https://github.com/beautify-web/js-beautify/issues/842

### CODE REVIEW

**Comment by HanabishiRecca** [comment]
@bitwiseman, check this.

File: js/src/css/beautifier.js [hunk 1]
```diff
@@ -334,7 +334,7 @@ Beautifier.prototype.beautify = function() {
         }
       }
     } else if (this._ch === ":") {
-      if ((insideRule || enteringConditionalGroup) && !(this._input.lookBack("&") || this.foundNestedPseudoClass()) && !this._input.lookBack("(") && !insideAtExtend) {
+      if ((insideRule || enteringConditionalGroup) && !(this._input.lookBack("&") || this.foundNestedPseudoClass()) && !this._input.lookBack("(") && !insideAtExtend && parenLevel < 1) {
```
Comment (bitwiseman) [inline js/src/css/beautifier.js]:
```suggestion
      if ((insideRule || enteringConditionalGroup) && !(this._input.lookBack("&") || this.foundNestedPseudoClass()) && !this._input.lookBack("(") && !insideAtExtend && parenLevel === 0) {
```

File: js/src/css/beautifier.js [hunk 2]
```diff
@@ -366,21 +366,27 @@ Beautifier.prototype.beautify = function() {
...
-      this.print_string(this._ch);
-      this.eatWhitespace(true);
-
-      // This maintains single line comments on the same
-      // line. Block comments are also affected, but
-      // a new line is always output before one inside
-      // that section
-      if (this._input.peek() !== '/') {
-        this._output.add_new_line();
+      if (parenLevel < 1) {
```
[the remark this answers is not in the document]
Comment (HanabishiRecca) [reply js/src/css/beautifier.js]: @bitwiseman, I've got this construction from already existing https://github.com/beautify-web/js-beautify/blob/ed7138e598a962164609bd588cddf145cc627609/js/src/css/beautifier.js#L414 and https://github.com/beautify-web/js-beautify/blob/ed7138e598a962164609bd588cddf145cc627609/js/src/css/beautifier.js#L419
So, should we bring all of it to a such style?
But actually negative value can happen. Isn't it more safe than zero equality?

[the remark this answers is not in the document]
Comment (bitwiseman) [reply js/src/css/beautifier.js]: @HanabishiRecca 
😱 😭 
parenLevel should never be less than zero.  But I can see how the current code can set it to that.

**Comment by HanabishiRecca** [comment]
@bitwiseman, alright now 'parenLevel' can't be negative.

**Comment by bitwiseman** [comment]
@HanabishiRecca  
I added tests. One to verify the parenthesis change and one for indenting of parenthesis contents.

**Pull request merged**